tests/suites/test_suite_lmots.function

```diff
@@ -23,7 +23,7 @@ int check_lmots_private_key_for_leak(unsigned char * sig)
 /* END_HEADER */
 
 /* BEGIN_DEPENDENCIES
- * depends_on:MBEDTLS_LMS_C
+ * depends_on:MBEDTLS_LMS_C:PSA_WANT_ALG_SHA_256
```

Comment:
Since you can't turn on `MBEDTLS_LMS_C` without `PSA_WANT_ALG_SHA_256`, adding a dependency on `PSA_WANT_ALG_SHA_256` won't change anything. Or rather that's what should happen if not for a mistake I hadn't spotted until now in `check_config.h`.

For the failure of `test_depends_hashes`, you need to edit the Perl script to exclude LMS when SHA256 is disabled. I think this is right:

```
--- a/tests/scripts/depends-hashes.pl
+++ b/tests/scripts/depends-hashes.pl
@@ -57,7 +57,8 @@ my @hash_configs = (
     ['unset MBEDTLS_MD5_C'],
     ['unset MBEDTLS_SHA512_C', 'unset MBEDTLS_SHA384_C '],
     ['unset MBEDTLS_SHA384_C'],
-    ['unset MBEDTLS_SHA256_C', 'unset MBEDTLS_SHA224_C'],
+    ['unset MBEDTLS_SHA256_C', 'unset MBEDTLS_SHA224_C',
+     'unset MBEDTLS_LMS_C'],
     ['unset MBEDTLS_SHA1_C'],
 );
 
```